netperf/archive.go

```diff
@@ -1,36 +1,119 @@
 package netperf
 
 import (
+	"context"
+	"crypto/tls"
 	"encoding/csv"
+	"encoding/json"
 	"fmt"
+	"net/http"
 	"os"
 	"strconv"
+	"strings"
 	"time"
 
-	"github.com/elastic/go-elasticsearch/v8"
+	"gihub.com/jtaleric/k8s-netperf/logging"
+	"gihub.com/jtaleric/k8s-netperf/metrics"
+	"github.com/google/uuid"
+	"github.com/opensearch-project/opensearch-go"
+	"github.com/opensearch-project/opensearch-go/opensearchapi"
 )
 
-type elasticParams struct {
-	url      string
-	user     string
-	password string
-	index    string
+const index = "k8s-netperf"
+
+// Doc struct of the JSON document to be indexed
+type Doc struct {
+	UUID          string           `json:"uuid"`
+	Timestamp     time.Time        `json:"timestamp"`
+	HostNetwork   bool             `json:"hostNetwork"`
+	Parallelism   int              `json:"parallelism"`
+	Profile       string           `json:"profile"`
+	Duration      int              `json:"duration"`
+	Samples       int              `json:"samples"`
+	Messagesize   int              `json:"messageSize"`
+	Result        float64          `json:"result"`
+	Metric        string           `json:"metric"`
+	ServerNodeCPU metrics.NodeCPU  `json:"serverCPU"`
+	ServerPodCPU  []metrics.PodCPU `json:"serverPods"`
+	ClientNodeCPU metrics.NodeCPU  `json:"ClientCPU"`
+	ClientPodCPU  []metrics.PodCPU `json:"ClientPods"`
```

Comment:
Just to follow the camelCase semanthics maybe we can rename these JSON fields?
```suggestion
	ClientNodeCPU metrics.NodeCPU  `json:"clientCPU"`
	ClientPodCPU  []metrics.PodCPU `json:"clientPods"`
```